setup.py

```diff
@@ -213,8 +229,48 @@ def check_linker_need_libatomic():
 CYTHON_HELPER_C_FILES = ()
 
 CORE_C_FILES = tuple(grpc_core_dependencies.CORE_SOURCE_FILES)
+excluded_core_c_files = []
+if "linux" in sys.platform:
+    CORE_C_FILES = filter(lambda x: 'third_party/libuv/src/win' not in x,
+                          CORE_C_FILES)
+    # The following source files are under libuv/src/unix but only for Darwin.
+    # They need to be excluded when running on Linux, and must keep synchronous
+    # with darwin_uv_srcs in src/libuv/gen_build_yaml.py, except
```

Comment:
:-( 
good luck maintaining this and gen_build_yaml.py
Now we have like 3 sources of truth for libuv sources on different platforms.
If python owners are ok with this, I'm "OK" with this too, but please note that it's very dirty and I explicitly refuse to maintain this in the future.